### PR TITLE
SCMOD-15888: Add constant

### PR DESCRIPTION
**Pull request by xbreizh**
https://portal.digitalsafe.net/browse/SCMOD-15888

### CODE REVIEW

**Comment by xbreizh** [comment]
@dermot-hardy Hi, do we want any form of validation for the constants? Are we using it solely to store functions and if so, should we keep that name or change it for something like `functions` for example?

**Comment by dermot-hardy** [comment]
No validation is required.  We'll probably only use it for functions but in principle it could be used for anything, so no don't change it; stick with that name.

File: job-service/src/main/java/com/hpe/caf/services/job/jobtype/JsltTaskBuilder.java [hunk 1]
```diff
@@ -58,12 +60,14 @@
...
         final String jobTypeId,
         final Map<String, String> configuration,
+        final Map<String, String> constants,
         final ParametersValidator parametersValidator,
         final String taskScript
     ) throws InvalidJobTypeDefinitionException {
         this.jobTypeId = jobTypeId;
         this.configuration = configuration;
         this.parametersValidator = parametersValidator;
+        this.constants = constants;
```
Comment (dermot-hardy) [inline job-service/src/main/java/com/hpe/caf/services/job/jobtype/JsltTaskBuilder.java]: Keep to initialize variables in the same order that you've listed the instance variables.

```suggestion
        this.constants = constants;
        this.parametersValidator = parametersValidator;
```

File: job-service/src/main/java/com/hpe/caf/services/job/jobtype/JsltTaskBuilder.java [hunk 2]
```diff
@@ -50,6 +51,7 @@
      * @param jobTypeId Job type's ID, used in error messages
      * @param configuration object containing string values which are fixed for the job type;
      *                      generally obtained from global configuration
+     * @param constants object containing string values which are fixed for the job type;
```
Comment (dermot-hardy) [inline job-service/src/main/java/com/hpe/caf/services/job/jobtype/JsltTaskBuilder.java]:
```suggestion
     * @param constants object containing string values which are fixed for the job type
```

**Review by dermot-hardy — APPROVED** [review]

File: job-service/src/test/java/com/hpe/caf/services/job/jobtype/DefaultDefinitionParserTest.java [hunk 3]
```diff
@@ -15,25 +15,26 @@
...
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.node.IntNode;
+import com.fasterxml.jackson.databind.node.NullNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.hpe.caf.services.configuration.AppConfig;
+import com.hpe.caf.services.job.api.generated.model.WorkerAction;
+import com.hpe.caf.services.job.exceptions.BadRequestException;
```
Comment (dermot-hardy) [inline job-service/src/test/java/com/hpe/caf/services/job/jobtype/DefaultDefinitionParserTest.java]: Remove the unnecessary re-ordering of the imports.

File: job-service/src/test/java/com/hpe/caf/services/job/jobtype/JsltTaskBuilderTest.java [hunk 4]
```diff
@@ -51,14 +53,14 @@ public void testBuild() throws Exception {
     @Test(expected = InvalidJobTypeDefinitionException.class)
     public void testBuildWithIncorrectScriptSyntax() throws Exception {
         new JsltTaskBuilder(
-            "type", Collections.emptyMap(), paramValidatorSuccess,
+            "type", Collections.emptyMap(), Collections.emptyMap(),paramValidatorSuccess,
```
Comment (dermot-hardy) [inline job-service/src/test/java/com/hpe/caf/services/job/jobtype/JsltTaskBuilderTest.java]:
```suggestion
            "type", Collections.emptyMap(), Collections.emptyMap(), paramValidatorSuccess,
```

File: job-service/src/test/resources/job-type-definitions/constant.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  - name: TARGET_PIPE
+constants:
+  anyScript: |-
+    function onProcessTask(e)
+    {
+       console.log('hello world!');
+    }
+taskScript: |
+  {
+  "taskClassifier": "classifier",
```
Comment (dermot-hardy) [inline job-service/src/test/resources/job-type-definitions/constant.yaml]:
```suggestion
    "taskClassifier": "classifier",
```

**Review by dermot-hardy — APPROVED** [review]

File: job-service/src/test/java/com/hpe/caf/services/job/jobtype/DefaultDefinitionParserTest.java (hunk 3 above)
Comment (xbreizh) [reply job-service/src/test/java/com/hpe/caf/services/job/jobtype/DefaultDefinitionParserTest.java]: Done

**Pull request merged**